src/fde.h

```diff
@@ -52,6 +52,9 @@ class fde
 {
 
 public:
+
+    static void Init();
```

Comment:
Missing doxygen comments for this public function. Simple as it looks there are complexities like the dependency that Squid_MaxFD being correct before this function is called, and that it asserts.

Reply:
> Missing doxygen comments for this public function.

Fixed in 3792efe:

```C++
// TODO: Merge with comm_init() to reduce initialization order dependencies.
/// Configures fd_table (a.k.a. fde::Table).
/// Call once, after learning the number of supported descriptors (i.e.
/// setMaxFD()) and before dereferencing fd_table (e.g., before Comm I/O).
static void Init();
```

> Also, we are using the Foo::Instance() pattern elsewhere to avoid globals. A TODO to mark this for future fix would be good.

The initialization complexities need to be hidden and better managed indeed -- I added a TODO -- but we probably should not use the singleton pattern _here_ because fd_table is not the only thing that needs proper (e.g., in the right moment and in the right order) configuration. We could, of course, move _all_ Comm accesses behind Comm::Instance(). Placing the whole module in a singleton _feels_ wrong to me now, but we should at least consider that idea when fixing Comm initialization.